### ISSUE TITLE
Line caps/joins

### ISSUE THREAD

**Issue by kaedroho** [ggez/ggez#545]
On an earlier unreleased version, you could write the following code to use custom line caps/joins:

```
graphics::set_line_join(ctx, graphics::LineJoin::Round);
graphics::set_line_cap(ctx, graphics::LineCap::Round);
graphics::line(…)
```

I can't find an equivalent that works on the latest version. These options now need to be specified on the ``DrawMode``, but it's not possible possible to specify a custom ``DrawMode`` when drawing lines.

**Comment by icefoxen** [comment]
Oops, you are correct!  These are properties on `CustomLine`, see here: https://docs.rs/lyon/0.11.0/lyon/tessellation/struct.StrokeOptions.html#structfield.end_cap  However, making them nicer properties either on `MeshBuilder` or in `DrawMode` (a la  #546) would be a good idea.

**Comment by icefoxen** [comment]
Fixed by #546 ; you can set these in `StrokeOption`, which is re-exported from `graphics::mesh`.
